src/sentry/releasehealth/metrics.py

```diff
@@ -423,4 +417,45 @@ def f(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
             query, referrer="releasehealth.metrics.check_has_health_data", use_cache=False
         )
 
-        return {extract_raw_info(raw) for raw in result["data"]}
+        return {extract_row_info(row) for row in result["data"]}
+
+    def check_releases_have_health_data(
+        self,
+        organization_id: OrganizationId,
+        project_ids: Sequence[ProjectId],
+        release_versions: Sequence[ReleaseName],
+        start: datetime,
+        end: datetime,
+    ) -> Set[ReleaseName]:
+
+        release_column_name = tag_key(organization_id, "release")
+        releases_ids = [
+            release_id
+            for release_id in [
+                try_get_tag_value(organization_id, release) for release in release_versions
+            ]
+            if release_id is not None
+        ]
+        query = Query(
+            dataset=Dataset.Metrics.value,
+            match=Entity("metrics_counters"),
+            select=[Column(release_column_name)],
+            where=[
+                Condition(Column("org_id"), Op.EQ, organization_id),
+                Condition(Column("project_id"), Op.IN, project_ids),
+                Condition(Column("metric_id"), Op.EQ, metric_id(organization_id, "session")),
+                Condition(Column(release_column_name), Op.IN, releases_ids),
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, end),
+            ],
+            groupby=[Column(release_column_name)],
+        )
+
+        result = raw_snql_query(
+            query, referrer="releasehealth.metrics.check_releases_have_health_data", use_cache=False
+        )
+
+        def extract_row_info(row: Mapping[str, Union[OrganizationId, str]]) -> ReleaseName:
+            return reverse_tag_value(organization_id, row.get(release_column_name))  # type: ignore
```

Comment:
Adding here a `# type: ignore` is unnecessary imo.. I'd much rather not have `ReleaseName` as a type and return a `str` here instead since `ReleaseName` is just basically an `str`

Reply:
Please `# type: ignore` comments should not be used liberally, and there must be a good reason to actually add them

Reply:
Sure, #type: ignore should not be used liberally, it is necessary here to avoid error generated by `row.get()`:

```
Argument 2 to "reverse_tag_value" has incompatible type "Union[int, str, None]"; expected "int"
```

Regarding the point on ReleaseName I disagree though. Any way ReleaseName is used in more APIs.